tests/suite-unit/PluginFlyvemdmPolicyRemoveapplication.php

```diff
@@ -0,0 +1,279 @@
+ *
+ * Flyve MDM Plugin for GLPI is a subproject of Flyve MDM. Flyve MDM is a mobile
+ * device management software.
+ *
+ * Flyve MDM Plugin for GLPI is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * Flyve MDM Plugin for GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Flyve MDM Plugin for GLPI. If not, see http://www.gnu.org/licenses/.
+ * ------------------------------------------------------------------------------
+ * @author    Domingo Oropeza
+ * @copyright Copyright © 2018 Teclib
+ * @license   AGPLv3+ http://www.gnu.org/licenses/agpl.txt
+ * @link      https://github.com/flyve-mdm/glpi-plugin
+ * @link      https://flyve-mdm.com/
+ * ------------------------------------------------------------------------------
+ */
+
+namespace tests\units;
+
+use Flyvemdm\Tests\CommonTestCase;
+
+class PluginFlyvemdmPolicyRemoveapplication extends CommonTestCase {
+
+   private $dataField = [
+      'group'     => 'application',
+      'symbol'    => 'removeApp',
+      'type_data' => '',
+      'unicity'   => 0,
+   ];
+
+   private $packageName = 'com.domain.author.application.apk';
+   private $filename = 'application.apk';
+
+   /**
+    * @param null|\PluginFlyvemdmPolicy $policyData
+    * @return array
+    */
+   private function createNewPolicyInstance($policyData = null) {
+      if (null === $policyData) {
+         $policyData = new \PluginFlyvemdmPolicy();
+         $policyData->fields = $this->dataField;
+      }
+      $policy = $this->newTestedInstance($policyData);
+      return [$policy, $policyData];
+   }
+
+   /**
+    * @return array
+    */
+   protected function integrityCheckProvider() {
+      return [
+         'Check ID exist'   => [
+            'data'     => ['', '', ''],
+            'expected' => [false, 'An application ID is required'],
+         ],
+         'Check ID exist 2' => [
+            'data'     => ['', '', 0],
+            'expected' => [false, 'An application ID is required'],
+         ],
+         'Check ID exist 3' => [
+            'data'     => ['', '', null],
+            'expected' => [false, 'An application ID is required'],
+         ],
+         'Valid check 1'    => [
+            'data'     => ['', '', 1],
+            'expected' => [true],
+         ],
+      ];
+   }
+
+   /**
+    * @dataProvider integrityCheckProvider
+    * @tags testIntegrityCheck
+    * @param array $data
+    * @param array $expected
+    */
+   public function testIntegrityCheck(array $data, array $expected) {
```

Comment:
Please split in the provider and here the argument $data into 3 variables with a meaningful name

Reply:
give subkeys to the key **data**

example:
```php
'data'     => ['package_name' => 'fake.lorem.ipsum.package', 'itemtype' => '', items_id => 0],
```

and the unit tests will be more readable